spec/lib/hubspot/config_spec.rb

```diff
@@ -1,52 +1,86 @@
 describe Hubspot::Config do
-  describe "#configure" do
-    let(:config){ {hapikey: "demo", base_url: "http://api.hubapi.com/v2", portal_id: "62515"} }
-    subject{ Hubspot::Config.configure(config) }
+  describe ".configure" do
+    it "sets the hapikey config" do
+      hapikey = "demo"
 
-    it "changes the hapikey config" do
-      expect{ subject }.to change(Hubspot::Config, :hapikey).to("demo")
+      config = Hubspot::Config.configure(hapikey: hapikey)
+
+      expect(config.hapikey).to eq(hapikey)
     end
 
     it "changes the base_url" do
-      expect{ subject }.to change(Hubspot::Config, :base_url).to("http://api.hubapi.com/v2")
+      base_url = "https://api.hubapi.com/v2"
+
+      config = Hubspot::Config.configure(
+        hapikey: "123abc",
+        base_url: base_url
+      )
+
+      expect(config.base_url).to eq(base_url)
     end
 
     it "sets a default value for base_url" do
-      Hubspot::Config.base_url.should == "https://api.hubapi.com"
+      config = Hubspot::Config.configure(hapikey: "123abc")
+
+      expect(config.base_url).to eq("https://api.hubapi.com")
     end
 
     it "sets a value for portal_id" do
-      expect{ subject }.to change(Hubspot::Config, :portal_id).to("62515")
+      portal_id = "62515"
+
+      config = Hubspot::Config.configure(
+        hapikey: "123abc",
+        portal_id: portal_id
+      )
+
+      expect(config.portal_id).to eq(portal_id)
+    end
+
+    it "raises when an authentication approach is not provided" do
+      expect {
+        Hubspot::Config.configure({})
+      }.to raise_error(Hubspot::ConfigurationError)
+    end
+
+    it "raises when two authentication approaches are provided" do
+      expect {
+        Hubspot::Config.configure({
+          hapikey: "123abc",
+          access_token: "456def",
+        })
+      }.to raise_error(Hubspot::ConfigurationError)
     end
   end
 
-  describe "#reset!" do
-    let(:config){ {hapikey: "demo", base_url: "http://api.hubapi.com/v2", portal_id: "62515"} }
-    before{ Hubspot::Config.configure(config) }
-    subject{ Hubspot::Config.reset! }
-    it "clears out the config" do
-      subject
-      Hubspot::Config.hapikey.should be_nil
-      Hubspot::Config.base_url.should == "https://api.hubapi.com"
-      Hubspot::Config.portal_id.should be_nil
+  describe ".reset!" do
+    it "resets the config values" do
+      Hubspot::Config.configure(hapikey: "123abc", portal_id: "456def")
+
+      Hubspot::Config.reset!
+
+      expect(Hubspot::Config.hapikey).to be nil
+      expect(Hubspot::Config.portal_id).to be nil
     end
   end
 
-  describe "#ensure!" do
-    subject{ Hubspot::Config.ensure!(:hapikey, :base_url, :portal_id)}
-    before{ Hubspot::Config.configure(config) }
+  describe ".ensure!" do
```

Comment:
It wasn't necessary to update these tests but I opted to update them to remove the calls to `should` which prints a deprecation warning:
> Using `should` from rspec-expectations' old `:should` syntax without explicitly enabling the syntax is deprecated. Use the new `:expect` syntax or explicitly enable `:should` with `config.expect_with(:rspec) { |c| c.syntax = :should }` instead.